tests/cpgrid/lgr/LgrChecks.hpp

```diff
@@ -817,5 +825,32 @@ void Opm::checkGlobalActiveCellsCountInGridWithLgrs(const Dune::CpGrid& grid,
 }
 
 
+void Opm::checkMarksAfterPreAdapt(const Dune::CpGrid& grid,
+                                  bool preAdapt)
+{
+    if (preAdapt) { // at least one element has been marked
+        for (const auto& element : Dune::elements(grid.leafGridView())) {
+            if (grid.getMark(element) == 1) { 
+                BOOST_CHECK(element.mightVanish()); // might disappear during the next call to adapt().
+            }
+            else {
+                BOOST_CHECK(element.isLeaf());
+            }
+        }
+    }
+}
+void Opm::checkMarksAfterPostAdapt(const Dune::CpGrid& grid,
+                                   int preAdaptMaxLevel,
+                                   bool isNested)
+{
+    for (const auto& element : Dune::elements(grid.leafGridView())) {
+        if (!isNested && (element.level() > preAdaptMaxLevel)) { // born in last refinement call
```

Comment:
Concening the `!isNested`:I think we should not skip checks in certain situations.

Maybe there is a way to rewrite this? If we know which elements are supposed to have gotten refined then we can check that their children are new. 

There could be a functor that returns whether an element was supposed to be refined.